lib/core.js

```diff
@@ -295,18 +295,26 @@ declare class JSON {
 }
 
 /* Iterators */
-interface IteratorResult<T> {
+interface IteratorResult<Y,R> {
```

Comment:
Why did you decide to parameterize `IteratorResult` on two things rather than just one? They seem equivalent to me...I'm just curious